src/dwcahandler/dwca/__init__.py

```diff
@@ -176,8 +181,43 @@ class Defaults:
 
 
 # Imports at end of file to allow classes to be used
-from dwcahandler.dwca.terms import Terms
-from dwcahandler.dwca.dwca_meta import Element, MetaElementTypes, MetaElementInfo, MetaDwCA
+from dwcahandler.dwca.terms import Terms, NsPrefix
+from dwcahandler.dwca.dwca_meta import (MetaElementTypes, MetaElementInfo, MetaDwCA,
+                                        MetaElementAttributes, get_meta_class_row_type)
+@dataclass
+class CsvFileType:
```

Comment:
Maybe we define something like TabularDataType which has `tables` instead of `files` and `CsvFileType` can be one of the table elements with the CSV encoding in there. That way you can have multiple CSVs with different CSV Encodings. 
To be more clear each `table` can be a `TableType` with attributes like `source data` which can be one of [`CSVFileType`, `pf.DataFrame`, `TextIOWrapper`], I guess `type`, `keys` can stay in `TabularDataType`.